### PR TITLE
bug fix read lon/lat

### PR DESCRIPTION
**Pull request by raphaeldussin**
Well this is embarrassing and quite urgent. Looks like PR #49 broke my regridding of MOM6 data.
The default behavior is now a breaking change and I did not catch it in my review. I strongly think
that the default behavior should mimic the old behavior (i.e. looking for `lon` and `lat` in dataset).

This PR reintroduce the old behavior and only switch to using cf-xaray if `lon` and `lat` are not present.
I also added a realistic test case using MOM6 ocean data pulled from opendap to avoid another bad surprise
in the future.

@huard @aulemahal I'd like us to move quickly on this and we probably need to push another release soon too!

### CODE REVIEW

File: xesmf/tests/test_oceanmodels.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import numpy as np
+import xarray as xr
+
+import xesmf
+
+
+def test_MOM6_to_1x1():
+    """ check regridding of MOM6 to regular lon/lat """
+
+    dataurl = 'http://35.188.34.63:8080/thredds/dodsC/OM4p5/'
```
Comment (huard) [inline xesmf/tests/test_oceanmodels.py:11]: Not sure I like the idea of streaming data from a THREDDS server in the test suite. We're doing this on our end and its prone to breakage, network issues, etc. I suggest creating a similar, small synthetic test file on the fly.

**Review by huard — APPROVED** [review]
Is this something that can be fixed later in cf-xarray ? Is this because cf-xarray is not aware of the different grid cell coordinates (tracer, u, v, etc) ?

**Review by aulemahal — APPROVED** [review]
Oh! I didn't think of that issue. The only problem with having 2 "longitude" happened to me because of #55, but I forgot ocean models (and surely others) often have more than one equally appropriate lon/lat pairs.
Overriding inference by name-matching sounds fine to me.

File: xesmf/tests/test_oceanmodels.py (hunk 1 above)
Comment (raphaeldussin) [reply xesmf/tests/test_oceanmodels.py:13]: fair point! I made that test optional for the CI but I'd like to keep the possibility to do regression tests against real data.

**Comment by slevang** [comment]
This change broke one of my workflows using DataArrays that did not have CF-compliant attributes, as the "old way" fix only worked for Datasets. PR #73 fixes this by switching to the `.coords` method to check for lat/lon fields.

**Pull request merged**